src/sentry/static/sentry/app/components/buttons/button.jsx

```diff
@@ -90,70 +74,170 @@ class Button extends React.Component {
 
-    // Props common to all elements
-    let componentProps = {
-      disabled,
-      ...buttonProps,
-      onClick: this.handleClick,
-      className: cx,
-      role: 'button',
-      children: childContainer,
-    };
-
-    // Handle react-router Links
-    if (to) {
-      return <Link to={this.getUrl()} {...componentProps} />;
-    }
+    return (
+      <StyledSmartButton
+        to={this.getUrl(to)}
+        href={this.getUrl(href)}
+        size={size}
+        {...buttonProps}
+        onClick={this.handleClick}
+        className={classNames(cxs)}
+        role="button"
+      >
+        <ButtonLabel size={size}>
+          {icon && (
+            <Icon size={size} hasChildren={!!children}>
+              <StyledInlineSvg src={icon} size={size === 'small' ? '12px' : '16px'} />
+            </Icon>
+          )}
+          {children}
+        </ButtonLabel>
+      </StyledSmartButton>
+    );
+  }
+}
 
-    if (href && external) {
-      return <ExternalLink href={this.getUrl()} {...componentProps} />;
-    }
+export default Button;
+
+const getFontSize = ({size}) => {
+  switch (size) {
+    case 'xsmall':
+    case 'small':
+      return '12px';
+    case 'large':
+      return '16px';
+    default:
+      return '14px';
+  }
+};
+
+const getFontWeight = ({priority}) => `font-weight: ${priority === 'link' ? 400 : 600};`;
+
+const getBoxShadow = active => ({priority, borderless, disabled}) => {
+  if (disabled || borderless || priority === 'link') {
+    return 'box-shadow: none';
+  }
 
-    // Handle traditional links
-    if (href) {
-      return <a href={this.getUrl()} {...componentProps} />;
+  return `box-shadow: ${active ? 'inset' : ''} 0 2px rgba(0, 0, 0, 0.05)`;
+};
+
+const getColors = ({priority, disabled, theme}) => {
+  let themeName = disabled ? 'disabled' : priority || 'default';
+  let {
+    color,
+    colorActive,
+    background,
+    backgroundActive,
+    border,
+    borderActive,
+  } = theme.button[themeName];
+
+  return css`
+    color: ${color};
+    background: ${background};
+    border: ${border ? `1px solid ${border}` : 'none'};
+
+    &:hover,
+    &:focus,
+    &:active {
+      ${colorActive ? 'color: ${colorActive};' : ''};
+      background: ${backgroundActive};
+      border: ${borderActive ? `1px solid ${borderActive}` : 'none'};
     }
+  `;
+};
+
+const StyledSmartButton = styled(({busy, external, ...props}) => {
+  // Get component to use based on existance of `to` or `href` properties
+  // Can be react-router `Link`, `a`, or `button`
+  if (props.to) {
+    return <Link {...props} />;
+  }
 
-    // Otherwise, fall back to basic button element
-    return <button {...componentProps} />;
+  if (!props.href) {
+    return <button {...props} />;
   }
-}
 
-export default Button;
+  let Component = external ? ExternalLink : 'a';
+  return <Component {...props} />;
+})`
+  display: inline-block;
+  line-height: 1;
+  border-radius: 3px;
```

Comment:
use `${p.theme.borderRadius}`

Reply:
I made a new theme variable since the old button radius = 3px (and theme is 4px). This way it's easier to spot regressions with percy. Let's create a followup PR if we do want to change the radius.